gridstatus/ercot.py

```diff
@@ -4491,6 +4491,9 @@ def get_mcpc_sced(
             published_before = end
             published_after = date
 
+            # Since we are filtering by publish before and end, set date to None
+            date = None
```

Comment:
I'm trying to think if this is the cleanest fix and if there is a fix that would avoid such issues in the future. 

do we ever **_need_** to provide both `date` and `publish_time_before`/`after` to `_get_documents`?

the bug here is because they both do different filtering on the same property and combine in unintuitive ways.

https://github.com/gridstatus/gridstatus/blob/745e2c6faf9dddcc1c6331d7f058f145081e9cc8/gridstatus/ercot.py#L3881-L3886

https://github.com/gridstatus/gridstatus/blob/745e2c6faf9dddcc1c6331d7f058f145081e9cc8/gridstatus/ercot.py#L3901-L3904

it seems like we could raise an error if both are provided together since I can't come up with a reason where that would be necessary

Reply:
I see only 4 places where we provide both `date` and `published_before`/`after` to _get_documents. it's only these new method. i'd think it's safe to raise an error without breaking any existing code

<img width="920" height="874" alt="Image" src="https://github.com/user-attachments/assets/f33824a1-9529-49f7-98cd-fea1e237b698" />

Reply:
Yeah, I thought about changing `_get_documents` but decided against it because it's called in many other places.

Reply:
Even if we make that change to `_get_documents`, we still have to make the change to this method to set `date=None`.

Reply:
is there a way to get this method to never pass date to get_documents?